x-pack/test_serverless/api_integration/test_suites/observability/ai_assistant/tests/public_complete/public_complete.spec.ts

```diff
@@ -0,0 +1,316 @@
+      conversationSimulator: LlmResponseSimulator
+    ) => Promise<void>;
+
+    async function getResponseBody(
+      { actions, instructions, format = 'default' }: RequestOptions,
+      conversationSimulatorCallback: ConversationSimulatorCallback
+    ) {
+      const titleInterceptor = proxy.intercept('title', (body) => isFunctionTitleRequest(body));
+
+      const conversationInterceptor = proxy.intercept(
+        'conversation',
+        (body) => !isFunctionTitleRequest(body)
+      );
+
+      const responsePromise = observabilityAIAssistantAPIClient.slsUser({
+        endpoint: 'POST /api/observability_ai_assistant/chat/complete 2023-10-31',
+        roleAuthc,
+        internalReqHeader,
+        params: {
+          query: { format },
+          body: {
+            messages,
+            connectorId,
+            persist: true,
+            actions,
+            instructions,
+          },
+        },
+      });
+
+      const [conversationSimulator, titleSimulator] = await Promise.race([
+        Promise.all([
+          conversationInterceptor.waitForIntercept(),
+          titleInterceptor.waitForIntercept(),
+        ]),
+        // make sure any request failures (like 400s) are properly propagated
+        responsePromise.then(() => []),
+      ]);
+
+      await titleSimulator.status(200);
+      await titleSimulator.next('My generated title');
+      await titleSimulator.complete();
+
+      await conversationSimulator.status(200);
+      if (conversationSimulatorCallback) {
+        await conversationSimulatorCallback(conversationSimulator);
+      }
+
+      const response = await responsePromise;
+
+      return String(response.body);
+    }
+
+    async function getEvents(
+      options: RequestOptions,
+      conversationSimulatorCallback: ConversationSimulatorCallback
+    ) {
+      const responseBody = await getResponseBody(options, conversationSimulatorCallback);
+
+      return responseBody
+        .split('\n')
+        .map((line) => line.trim())
+        .filter(Boolean)
+        .map((line) => JSON.parse(line) as StreamingChatResponseEvent)
+        .slice(2); // ignore context request/response, we're testing this elsewhere
+    }
+
+    async function getOpenAIResponse(conversationSimulatorCallback: ConversationSimulatorCallback) {
+      const responseBody = await getResponseBody(
+        {
+          format: 'openai',
+        },
+        conversationSimulatorCallback
+      );
+
+      return responseBody;
+    }
+
+    before(async () => {
+      proxy = await createLlmProxy(log);
+      roleAuthc = await svlUserManager.createM2mApiKeyWithRoleScope('admin');
+      internalReqHeader = svlCommonApi.getInternalRequestHeader();
+      connectorId = await createProxyActionConnector({
+        supertest,
+        log,
+        port: proxy.getPort(),
+        internalReqHeader,
+        roleAuthc,
+      });
+    });
+
+    after(async () => {
+      await deleteAllConversations({
```

Comment:
added this here because the `with openai format` test suite looks to be creating conversation and not cleaning up.  the response does not look to return a conversation id.